Framework/PythonInterface/plugins/algorithms/ReflectometrySliceEventWorkspace.py

```diff
@@ -83,10 +90,93 @@ def _slice_input_workspace(self):
-                             LogText=str(run_number))
+            self._copy_run_number_to_sample_log(ws, ws)
+        return group
+
+    def _create_filter(self):
+        """Generate the splitter workspace for performing the filtering for each required slice"""
+        alg = self.createChildAlgorithm("GenerateEventsFilter")
+        for property_name in self._filter_properties:
+            alg.setProperty(property_name, self.getPropertyValue(property_name))
+        alg.setProperty("OutputWorkspace", '__split')
+        alg.setProperty("InformationWorkspace", '__info')
+        alg.execute()
+        self._split_ws = alg.getProperty("OutputWorkspace").value
+        self._info_ws = alg.getProperty("InformationWorkspace").value
+
+    def _slice_input_workspace_with_filter_by_time(self):
+        # Get the start/stop times, or use the run start/stop times if they are not provided
+        run_start = DateAndTime(self._input_ws.run().startTime())
+        run_stop = DateAndTime(self._input_ws.run().endTime())
+        start_time = self._get_property_or_default_as_datetime("StartTime", default_value=run_start,
+                                                               relative_start=run_start)
+        stop_time = self._get_property_or_default_as_datetime("StopTime", default_value=run_stop,
+                                                              relative_start=run_start)
+        # Get the time interval, or use the total interval if it's not provided
+        total_interval = (stop_time - start_time).total_seconds()
+        time_interval = self._get_interval_as_float("TimeInterval", total_interval)
+        # Calculate start/stop times in seconds relative to the start of the run
+        relative_start_time = (start_time - run_start).total_seconds()
+        relative_stop_time = relative_start_time + total_interval
+        # Loop through each slice
+        slice_names = list()
+        slice_start_time = relative_start_time
+        while slice_start_time < relative_stop_time:
+            slice_stop_time = slice_start_time + time_interval
+            slice_name = self._output_ws_group_name + '_' + str(slice_start_time) + '_' + str(slice_stop_time)
+            slice_names.append(slice_name)
+            alg = self.createChildAlgorithm("FilterByTime")
+            alg.setProperty("InputWorkspace", self._input_ws)
+            alg.setProperty("OutputWorkspace", slice_name)
+            alg.setProperty("StartTime", str(slice_start_time))
+            alg.setProperty("StopTime", str(slice_stop_time))
+            alg.execute()
+            sliced_workspace = alg.getProperty("OutputWorkspace").value
+            mtd.addOrReplace(slice_name, sliced_workspace)
+            # Proceed to the next interval
+            slice_start_time = slice_stop_time
+        # Group the sliced workspaces
+        group = self._group_workspaces(slice_names, self._output_ws_group_name)
+        mtd.addOrReplace(self._output_ws_group_name, group)
```

Comment:
Would it be possible to add this workspace to the ADS using the algorithm output properties method? As I find that manually adding algorithms to the ADS as a side effect during algorithms can cause confusion if you later try and use the algorithm as a child.

Reply:
Although this may be beyond the scope of this PR

Reply:
Yes, beyond the scope for this PR but I've added a note to #27697 to do this because I'm doing a similar fix for other algorithms in that PR and I'd missed this one!

Reply:
Yes, beyond the scope for this PR but I've added a note to do this in #27697 because I'm fixing similar problems in our other algorithms in that PR and had missed this one!